lib/proto/proto.go

```diff
@@ -396,6 +396,7 @@ func setField(msg protoreflect.Message, fdesc protoreflect.FieldDescriptor, valu
 		defer iter.Done()
 
 		// TODO(adonovan): handle maps
+		msg.Clear(fdesc)
 		list := msg.Mutable(fdesc).List()
```

Comment:
list.Truncate(0) may avoid an allocation when the existing array is sufficient.

Reply:
Thanks - switched to list.Truncate(0).